app/controllers/candidate_interface/invites_controller.rb

```diff
@@ -2,19 +2,80 @@ module CandidateInterface
   class InvitesController < CandidateInterfaceController
     before_action :redirect_to_post_offer_dashboard_if_accepted_deferred_or_recruited
     before_action :redirect_if_feature_off_and_no_submitted_application
+    before_action :set_invite, only: %i[show decline update update_reason course_unavailable]
 
     def index
       @invites = current_application.published_invites.includes(:application_choice)
     end
 
     def show
-      @invite = Pool::Invite.find(params[:id])
+      if !@invite.course.open?
+        redirect_to course_unavailable_candidate_interface_invite_path(@invite)
+        return
+      end
+
+      @fac_invite_response_form = CandidateInterface::FacInviteResponseForm.new(
+        application_form: current_application,
+        invite: @invite,
+      )
+    end
+
+    def update
+      @fac_invite_response_form ||= CandidateInterface::FacInviteResponseForm.new(fac_invite_response_form_params.merge(
+                                                                                    application_form: current_application,
+                                                                                    invite: @invite,
+                                                                                  ))
+      if @fac_invite_response_form.valid?
+        if @fac_invite_response_form.accepted_invite?
+          @fac_invite_response_form.save
+          redirect_to candidate_interface_course_choices_course_review_path(@invite.application_choice.id)
+        else
+          redirect_to decline_candidate_interface_invite_path
+        end
+      else
+        track_validation_error(@fac_invite_response_form)
+        render :show
+      end
+    end
+
+    def decline
+      @fac_invite_decline_reason_form = CandidateInterface::FacInviteDeclineReasonsForm.new
+    end
+
+    def update_reason
+      @fac_invite_decline_reason_form = CandidateInterface::FacInviteDeclineReasonsForm.new(fac_invite_decline_reason_form_params)
+
+      if @fac_invite_decline_reason_form.save(@invite)
+        flash[:success] = [t('.header', course: @invite.course.name_and_code,
+                                        provider: @invite.provider_name),
+                           t('.body', link: view_context.govuk_link_to('apply to this course', candidate_interface_course_choices_course_confirm_selection_path(@invite.course)))]
+
+        redirect_to candidate_interface_invites_path
+      else
+        render :decline
+      end
```

Comment:
We should have a DeclineReasonsController where this logic lives. That way we can stick with standard crud methods. I know that isn't the way it has always been done in the past, but as a team, we have decided that we want to be predictably rails-y